Xamarin.Forms.Platform.WinRT/Platform.cs

```diff
@@ -747,8 +747,35 @@ void OnPageActionSheet(Page sender, ActionSheetArguments options)
 				dialog.CancelCommandIndex = (uint)dialog.Commands.Count - 1;
 			}
 
-			IUICommand command = await dialog.ShowAsync();
+			IUICommand command = await dialog.ShowAsyncQueue();
 			options.SetResult(command.Label == options.Accept);
 		}
 	}
+	
+	// refer to http://stackoverflow.com/questions/29209954/multiple-messagedialog-app-crash for why this is used
+	// in order to allow for multiple MessageDialogs, or a crash occurs otherwise
+	public static class MessageDialogExtensions
+	{
+		static TaskCompletionSource<MessageDialog> _currentDialogShowRequest;
+
+		public static async Task<IUICommand> ShowAsyncQueue(this MessageDialog dialog)
+		{
+			if (!Window.Current.Dispatcher.HasThreadAccess)
+			{
+				throw new InvalidOperationException("This method can only be invoked from UI thread.");
+			}
+
+			while (_currentDialogShowRequest != null)
+			{
+				await _currentDialogShowRequest.Task;
+			}
```

Comment:
That looks strange to me. As the signature is not `async void` the control won't return to the caller while this call is awaiting.

I'd rewrite this method to be `public static Task<IUICommand> ShowAsyncQueue(this MessageDialog dialog)` and fix this...